src/reducers/entities/channels.ts

```diff
@@ -65,9 +65,9 @@ function channels(state: IDMappedObjects<Channel> = {}, action: GenericAction) {
     case SchemeTypes.RECEIVED_SCHEME_CHANNELS: {
         const nextState = {...state};
 
-        for (const channel of action.data) {
-            if (state[channel.id] && channel.type === General.DM_CHANNEL) {
-                channel.display_name = channel.display_name || state[channel.id].display_name;
+        for (let channel of action.data) {
+            if (state[channel.id] && channel.type === General.DM_CHANNEL && !channel.display_name) {
+                channel = {...channel, display_name: state[channel.id].display_name};
```

Comment:
Do you know when we're dispatching a RECEIVED_CHANNELS action with data that's already in the store? That's weird to me that we can't mutate `action.data` because it's something that came from the store previously.

Reply:
I think Redux requires you not to modify any of the incoming parameters, including the action (https://redux.js.org/troubleshooting/#never-mutate-reducer-arguments):
> Never mutate reducer arguments
> It is tempting to modify the state or action passed to you by Redux. Don't do this!

Reply:
Oh, yeah. That makes sense now that you mention it since the action will also be handled by other reducers as well.